source/sphinx_extensions/command_block/extension.py

```diff
@@ -128,13 +139,26 @@ def run(self):
     def _get_env(self):
         return self.state.document.settings.env
 
-    def _get_literal_block_node(self, commands):
+    def _get_literal_block_node(self, commands, runtime):
         content = '\n'.join(commands)
         node = docutils.nodes.literal_block(content, content)
-        node['language'] = 'shell'
+        node['language'] = runtime
         return node
 
-    def _execute_commands(self, commands, working_dir):
+    def _execute_commands(self, commands, working_dir, runtime):
+        return {
+            'shell': self._execute_sh_commands,
+            'python': self._execute_py_commands,
+        }[runtime](commands, working_dir)
+
+    def _execute_py_commands(self, commands, working_dir):
+        with tempfile.NamedTemporaryFile(mode='w', encoding='utf-8') as fh:
+            for command in commands:
+                fh.write('%s\n' % command)
+            return self._execute_sh_commands(
+                ['%s %s' % (sys.executable, str(fh.name))], working_dir)
```

Comment:
Is there ever a need to share scope across blocks for this?

Reply:
Probably, and that is certainly cumbersome w this current approach (would require Artifacts to be saved to disk).